### ISSUE TITLE
Relationships between interfaces, ports, and transceivers may be confused

### ISSUE THREAD

**Issue by nkitchen** [openconfig/public#189]
openconfig-platform-port.yang augments /interfaces/interface/state with a leafref that looks like it should point to a component of type PORT:

<pre><code>augment "/oc-if:interfaces/oc-if:interface/oc-if:state" {
  description
    "Adds a reference from the base interface to the corresponding
    <b>port component</b> in the device inventory.";

  leaf hardware-port {
    type leafref {
      path "/oc-platform:components/oc-platform:component/" +
        "oc-platform:name";
    }
    description
      "For non-channelized interfaces, references the hardware <b>port</b>
      corresponding to the base interface.";
  }
}</code></pre>

However, openconfig-platform-transceiver.yang has another augment to /interfaces/interface/state that uses hardware-port and assumes that it points to a component with a child `transceiver`:

<pre><code>augment "/oc-if:interfaces/oc-if:interface/oc-if:state" {
  [...]
  leaf-list physical-channel {
    type leafref {
      path "/oc-platform:components/" +
        "oc-platform:component[oc-platform:name=current()/../<b>oc-port:hardware-port</b>]/" +
        "<b>oc-transceiver:transceiver</b>/" +
        "oc-transceiver:physical-channels/oc-transceiver:channel/" +
        "oc-transceiver:index";
    }
    description
      "For a channelized interface, list of references to the
      physical channels (lanes) corresponding to the interface.";
  }
}</code></pre>

Only components of type TRANSCEIVER have `transceiver` children.

A possible resolution: Instead of taking the component for the `physical-channel` leafrefs from `hardware-port`, take it from the transceiver subcomponent of `hardware-port`:

In openconfig-platform-transceiver.yang:
```
augment "/oc-if:interfaces/oc-if:interface/oc-if:state" {
  leaf transceiver {
    type leafref {
      path "/oc-platform:components/" +
        "oc-platform:component[oc-platform:name=current()/../oc-port:hardware-port]" +
        "oc-platform:subcomponents/oc-platform:subcomponent/oc-platform:name";
    }
  }
  
  leaf-list physical-channel {
    type leafref {
      path "/oc-platform:components/" +
        "oc-platform:component[oc-platform:name=current()/../oc-transceiver:transceiver]/" +
        "oc-transceiver:transceiver/" +
        "oc-transceiver:physical-channels/oc-transceiver:channel/" +
        "oc-transceiver:index";
    }
  }
}
```

**Comment by LeonGWang** [comment]
Seems like this is already resolved at some point: https://github.com/openconfig/public/blob/master/release/models/platform/openconfig-platform-transceiver.yang#L593

**Comment by nkitchen** [comment]
Fixed by 9d946210937527512c9fbb1576fdf0ad51e0d356
